library/psa_crypto_rsa.h

```diff
@@ -0,0 +1,136 @@
+/*
+ *  PSA RSA layer on top of Mbed TLS crypto
+ */
+/*
+ *  Copyright The Mbed TLS Contributors
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *  not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#ifndef PSA_CRYPTO_RSA_H
+#define PSA_CRYPTO_RSA_H
+
+#include <psa/crypto.h>
+#include <mbedtls/rsa.h>
+
+/** Load the contents of a key buffer into an internal RSA representation
+ *
+ * \param[in] type          The type of key contained in \p data.
+ * \param[in] data          The buffer from which to load the representation.
+ * \param[in] data_length   The size in bytes of \p data.
+ * \param[out] p_rsa        Returns a pointer to an RSA context on success.
+ *                          The caller is responsible for freeing both the
+ *                          contents of the context and the context itself
+ *                          when done.
+ */
+psa_status_t psa_crypto_rsa_load_representation( psa_key_type_t type,
```

Comment:
I can change this if you are convinced by an alternative. Not having `psa` in the function name open the door to name conflicts with the MbedTLS classic functions but that's maybe just a theoretical concern.

Reply:
We control the `mbedtls_` namespace, more so than the `psa_` namespace (even `psa_crypto_`. We can use `mbedtls_psa_` (which is informally growing for things that bridge the two APIs in either direction, as well as for pure-PSA things that are specific to Mbed TLS, but it isn't used systematically).

Reply:
ok thus the name of the file would be mbedtls_psa_rsa.c and the functions prefixed by mbedtls_psa_ ?

Reply:
The file name should stay `psa_xxx`. We have `rsa.c`, not `mbedtls_rsa.c`.

I currently favor function names like `mbedtls_psa_rsa_load_representation`. It would be good to discuss long-term plans at some point and establish clear naming conventions that are compatible with these long-term plans. But establishing these conventions is a longer-term thing than the driver testing work and should not hold it back.

Reply:
ok thus `psa_crypto_rsa.c` for the file name and `mbedtls_psa_rsa_xyz` for the function names, right?

Reply:
sounds good

Reply:
ok cool, I will go for that then, thanks.